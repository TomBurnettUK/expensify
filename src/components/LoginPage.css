.login-page {
  align-items: center;
  background: url("/images/bg.jpg");
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.login-box {
  background-color: rgba(255, 255, 255, .85);
  border-radius: 3px;
  padding: 2rem 1.5rem;
  text-align: center;
  width: 20rem;
}

.login-title {
  margin: 0;
  line-height: 1;
}

.login-description {
  margin: 1.5rem;
}

.login-button {
  background-color: #2F4F9F;
  color: #FFF;
  border: none;
  padding: 1rem 2rem;
  font-size: 1.1rem;
  font-weight: 300;
  line-height: 1;
}