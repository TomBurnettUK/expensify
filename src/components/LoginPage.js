import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

import './LoginPage.css';

export const LoginPage = props => (
  <div className="login-page">
    <div className="login-box">
      <h1 className="login-title">Expensify</h1>
      <p className="login-description">
        It&apos;s time to get your expenses under control!
      </p>
      <button className="login-button" onClick={props.startLogin}>
        Login with Google
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
