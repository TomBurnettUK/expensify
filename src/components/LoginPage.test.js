import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from './LoginPage';

it('should render correctly', () => {
  const wrapper = shallow(<LoginPage startLogin={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

it('should call startLogin on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});
