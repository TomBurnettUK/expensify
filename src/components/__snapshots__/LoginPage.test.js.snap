// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
